### DISCUSSION TITLE
[DropdownMenu]: Style applied to the `body[data-scroll-locked]` messes up `position: sticky` on the child elements

### DISCUSSION THREAD

**Discussion by arpitdalal** [radix-ui/primitives#2873] [Help]
Opening the Radix DropdownMenu makes a div with `position: sticky;` relative again for some reason. I wasn't able to reproduce this, so probably it's not a bug. But I am lost, where would the problem lie?

If I override the `overflow: hidden` behavior on body, then it works fine.

https://github.com/radix-ui/primitives/assets/61059807/c7526d40-a31e-4253-9156-9210ebc8cc81

**Comment by sasicodes** [comment]
+1

**Comment by lorekadam** [comment]
Hi, on the `DropdownRoot` try to set a `modal={false}` prop. It should help :)

**Reply by IgorKhomenko** [reply]
you saved my day.
I used https://ui.shadcn.com/docs/components/dropdown-menu and when trigger dropdown, the body becomes fully hidden in my app. Totally unclear...

**Reply by rishavanand** [reply]
Issue gets resolved with modal=false but that's only for dropdown. What if I want to use the modal itself with the backdrop? @lorekadam

**Comment by mikecreuzer** [comment]
+1

**Comment by Hexi1997** [comment]
+1

**Comment by abdulfattahaud** [comment]
+1
On the Popover component, it doesn't work, and neither does this `modal={false}`.

**Comment by braginteractive** [comment]
Anyone figure this out? Setting the modal to false doesnt seem to fix it either

**Comment by elanizi43998** [comment]
Same thing happened with me with the Sheet componet in shadcn

**Comment by theredled** [comment]
+1
